src/webui/www/private/scripts/client.js

```diff
@@ -88,11 +88,12 @@ const loadSelectedTracker = function() {
 loadSelectedTracker();
 
 function genHash(string) {
+    // origins:
+    // https://stackoverflow.com/a/8831937
+    // https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0
     let hash = 0;
-    for (let i = 0; i < string.length; ++i) {
-        const c = string.charCodeAt(i);
-        hash = (c + hash * 31) | 0;
-    }
+    for (let i = 0; i < string.length; ++i)
+        hash = ((Math.imul(hash, 31) + string.charCodeAt(i)) | 0);
```

Comment:
What does operator `|` mean in JS?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_OR

Reply:
But since this is just a `bitwise OR` (which I honestly began to doubt), isn't this part of expression meaningless? Isn't `(x | 0) == x` always true?

Reply:
https://gist.github.com/hyamamoto/fd435505d29ebfa3d9716fd2be8d42f0#gistcomment-2694461

Reply:
Perverted JavaScript...